src/main/java/emu/grasscutter/server/packet/send/PacketPlatformStartRouteNotify.java

```diff
@@ -0,0 +1,39 @@
+package emu.grasscutter.server.packet.send;
+
+import emu.grasscutter.game.entity.platform.EntityPlatform;
+import emu.grasscutter.game.world.Scene;
+import emu.grasscutter.net.packet.BasePacket;
+import emu.grasscutter.net.packet.PacketOpcodes;
+import emu.grasscutter.net.proto.MathQuaternionOuterClass;
+import emu.grasscutter.net.proto.PlatformInfoOuterClass;
+import emu.grasscutter.net.proto.PlatformStartRouteNotifyOuterClass;
+
+public class PacketPlatformStartRouteNotify extends BasePacket {
+    public PacketPlatformStartRouteNotify(int clientSequence, EntityPlatform entity, Scene scene) {
+        super(PacketOpcodes.PlatformStartRouteNotify, clientSequence);
+
+        var notify = PlatformStartRouteNotifyOuterClass.PlatformStartRouteNotify.newBuilder()
+            .setEntityId(entity.getId())
+            .setSceneTime(scene.getTime())
+            .setPlatform(PlatformInfoOuterClass.PlatformInfo.newBuilder()
+                .setStartSceneTime(scene.getTime())
+                .setIsStarted(true)
+                .setStartRot(MathQuaternionOuterClass.MathQuaternion.newBuilder()
+                    .setW(0.653F)
+                    .setY(0.757F)
+                    .build())
+                .setPosOffset(entity.getPosition().toProto())
+                .setRotOffset(MathQuaternionOuterClass.MathQuaternion.newBuilder()
+                    .setW(1.0F)
+                    .build())
+                .setMovingPlatformType(entity.getMovingPlatformType())
+                .setIsActive(true)
+                .build())
+            .build();
+
+        entity.setStarted(true);
```

Comment:
Wouldn't it be better to have something like entity.startRoute() that sets these things and sends this package, instead of letting the notifypackage set entity proppeties ?

Reply:
Agreed.  send Packets only have constructor(s) because they are used just as data for the server to send. Since entities are scene-bound objects, platforms should be managed by the scene they are in. This is especially important for seelie (they are considered platforms) which have complex routes and behaviour

Reply:
ok, I will fix it!